kyo-core/shared/src/main/scala/kyo/Hub.scala

```diff
@@ -165,33 +209,45 @@ object Hub:
       *   The result of applying the function
       */
     def initWith[A](capacity: Int)[B, S](f: Hub[A] => B < S)(using Frame): B < (S & IO) =
-        Channel.initWith[A](capacity) { ch =>
-            IO {
-                val listeners = new CopyOnWriteArraySet[Channel[A]]
-                Async.run {
-                    Loop.foreach {
-                        ch.take.map { v =>
-                            IO {
-                                val puts =
-                                    listeners.toArray()
-                                        .toList.asInstanceOf[List[Channel[A]]]
-                                        .map(child => Abort.run[Throwable](child.put(v)))
-                                Async.parallelUnbounded(puts).map(_ => Loop.continue)
-                            }
+        IO.Unsafe {
```

Comment:
Optimized to use unsafe operations to avoid allocations + improved error handling